hack/verify-gendocs.sh

```diff
@@ -39,20 +39,25 @@ if [[ ! -x "$gendocs" || ! -x "$genman" || ! -x "$genbashcomp" ]]; then
 fi
 
 DOCROOT="${KUBE_ROOT}/docs/"
-TMP_DOCROOT="${KUBE_ROOT}/docs_tmp/"
+TMP_DOCROOT="${KUBE_ROOT}/_tmp/docs/"
+_tmp="${KUBE_ROOT}/_tmp"
 
+mkdir -p "${_tmp}"
 cp -a "${DOCROOT}" "${TMP_DOCROOT}"
 
 kube::util::gen-doc "${genman}" "${TMP_DOCROOT}/man/man1/"
 kube::util::gen-doc "${gendocs}" "${TMP_DOCROOT}"
-
 echo "diffing ${DOCROOT} against freshly generated docs"
-set +e
-diff -Naupr -I 'Auto generated by' "${DOCROOT}" "${TMP_DOCROOT}"
+ret=0
+diff -Naupr -I 'Auto generated by' "${DOCROOT}" "${TMP_DOCROOT}" || ret=$?
 ret=$?
```

Comment:
I think this line should be dropped, otherwise you'll lose the error status since the last command will be `ret=$?` which will itself succeed!

```shell
$ ret=0; false || ret=$?; echo "${ret}"
1
$ ret=0; false || ret=$?; ret=$?; echo "${ret}"
0
```